salt/modules/cipher.py

```diff
@@ -0,0 +1,531 @@
+    return key
+
+
+def enc(data, **kwargs):
+    '''
+    alias for aes_enc
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt-call cipher.enc
+    '''
+    return aes_enc(data=data, **kwargs)
+
+
+def dec(data, **kwargs):
+    '''
+    alias for aes_dec
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt-call cipher.dec
+    '''
+    return aes_dec(data=data, **kwargs)
+
+
+def libnacl_keygen():
+    '''
+    Use libnacl to generate a private key
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt-call --out=newline_values_only cipher.libnacl_keygen > /root/.naclkey
+    '''
+    if REQ_ERROR['libnacl']:
+        raise Exception(REQ_ERROR['libnacl'])
+    b = libnacl.secret.SecretBox()
+    return base64.b64encode(b.sk)
+
+
+def libnacl_enc(data, **kwargs):
+    '''
+    Takes a key generated from `cipher.libnacl_keygen` and encrypt some data.
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt-call cipher.libnacl_enc datatoenc keyfile=/root/.naclkey
+    '''
+    if REQ_ERROR['libnacl']:
+        raise Exception(REQ_ERROR['libnacl'])
+    key = _get_key(**kwargs)
+    sk = base64.b64decode(key)
+    b = libnacl.secret.SecretBox(key=sk)
+    return base64.b64encode(b.encrypt(data))
+
+
+def libnacl_dec(data, **kwargs):
+    '''
+    Takes a key generated from `cipher.libnacl_keygen` and decrypt some data.
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt-call cipher.libnacl_dec pEXHQM6cuaF7A= keyfile=/root/.naclkey
+    '''
+    if REQ_ERROR['libnacl']:
+        raise Exception(REQ_ERROR['libnacl'])
+    key = _get_key(**kwargs)
+    sk = base64.b64decode(key)
+    b = libnacl.secret.SecretBox(key=sk)
+    return b.decrypt(base64.b64decode(data))
+
+
+def rsa_keygen(keysize=1024, passphrase=None):
+    '''
+    Use PyCrypto to generate a RSA PEM.
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt-call --out=newline_values_only cipher.rsa_keygen > /root/.rsakey
+        salt-call --out=newline_values_only cipher.rsa_keygen keysize=1024 > /root/.rsakey
+        salt-call --out=newline_values_only cipher.rsa_keygen passphrase=pp > /root/.rsakey
+    '''
+    if REQ_ERROR['crypto_rsa']:
+        raise Exception(REQ_ERROR['crypto_rsa'])
+    key = RSA.generate(keysize, os.urandom)
+    return key.exportKey('PEM', passphrase=passphrase)
+
+
+def rsa_enc(data, passphrase=None, **kwargs):
+    '''
+    Takes a key generated from `cipher.rsa_keygen` and encrypt data.
+
+    CLI Examples:
+
+    .. code-block:: bash
+
+        salt-call cipher.rsa_enc datatoenc keyfile=/root/.rsakey
+        salt-call cipher.rsa_enc datatoenc passphrase=pp keyfile=/root/.rsakey
+    '''
+    if REQ_ERROR['crypto_rsa']:
+        raise Exception(REQ_ERROR['crypto_rsa'])
+    key = _get_key(**kwargs)
+    #conf = _get_config(**kwargs)
+    rsakey = RSA.importKey(key, passphrase=passphrase)
+    rsapubkey = rsakey.publickey()
+    c = rsapubkey.encrypt(data, None)[0]
```

Comment:
I don't recall the exact syntax, but this needs to also include the padding class in pycrypto
Also, there is a bug in importkey in older pycrypto where the imported key would leave the cipher in plaintext